netbox_netprod_importer/devices_list.py

```diff
@@ -28,3 +31,57 @@ def parse_devices_yaml_def(devices_yaml, creds=None):
                     "Cannot connect to device %s: %s", hostname, e
                 )
     return devices
+
+
+def parse_filter_yaml_def(filter_yaml, creds=None):
+    netbox_api = NetboxAPI(**get_config()["netbox"])
+    devices = {}
+    with open(filter_yaml) as filter_yaml_str:
+        q_list = []
+        yml = yaml.safe_load(filter_yaml_str)
+        for fil, props in tqdm(yml["filter"].items()):
+            if props is not None:
+                if type(props) is list:
+                    q_list.append(fil + "=" +
+                                  ("&" + fil + "=").join(str(x) for x in props)
+                                  )
+                else:
+                    q_list.append(fil + "=" + str(props))
+        logger.info("Resultant filter %s", "&".join(q_list))
+        platforms_js = netbox_api.get("dcim/platforms")
+        platforms = {}
+        platforms_args = {}
+        for platform in platforms_js["results"]:
+            if platform["napalm_driver"]:
+                platforms[platform["id"]] = platform["napalm_driver"]
+                platforms_args[platform["id"]] = platform["napalm_args"]
+        if len(platforms) == 0:
+            logger.error("Not for one platform napalm_driver is not defined")
+            sys.exit(4)
+
+        devlist = netbox_api.get("dcim/devices/?" + "&".join(q_list))
+        for device in devlist["results"]:
+            if not device["platform"]:
+                continue
+            if not platforms.get(device["platform"]["id"]):
+                continue
+
+            try:
+                devices[device["name"]] = DeviceImporter(
+                    device["primary_ip"].get("address").split("/")[0]
```

Comment:
`device["primary_ip"].get("address").split("/")` might fail if the primary IP is not defined, as `device["primary_ip"].get("address")` default to `None`, and `None.split()` is not implemented.

Reply:
Corrected